### PR TITLE
Remove all vendoring

### PR DESCRIPTION
**Pull request by s-t-e-v-e-n-k**
With this project removing support for Python 2 as well as any version of Python 3 that is < 3.8, utilise pyupgrade to bring this project to that level of Python 3 code, as well as removing all vestiges of vendored backports.

### CODE REVIEW

**Comment by wbarnha** [comment]
I coincidentally ran `pyupgrade` in https://github.com/wbarnha/kafka-python-ng/pull/171 but I'll go ahead and review this to see what I missed.

**Comment by s-t-e-v-e-n-k** [comment]
I'm braced and ready for doing the rebase. :-)

**Comment by wbarnha** [comment]
> I'm braced and ready for doing the rebase. :-)

You responded as I was opening this PR. This is going to be a chunky one to rebase, I'll get around to merging it tomorrow

**Comment by s-t-e-v-e-n-k** [comment]
Yeah, I suspect it is going to be easier to carry over the vendor removal bits I want, and drop every other change. I can probably sort that out in the next 30 minutes if you wish.

**Comment by wbarnha** [comment]
> Yeah, I suspect it is going to be easier to carry over the vendor removal bits I want, and drop every other change. I can probably sort that out in the next 30 minutes if you wish.

If you could, that would be great! I'm pretty eager to get this merged.

**Comment by s-t-e-v-e-n-k** [comment]
Forced pushed, let's see if I missed anything. I'm not 100% happy with the crimes I committed in `benchmarks/variant_speed.py`, but I also didn't want to completely rewrite it to remove its reliance on the Python 2 construct of int2byte.

File: benchmarks/varint_speed.py [hunk 1]
```diff
@@ -67,6 +65,10 @@
 BENCH_VALUES_DEC = list(map(bytearray, BENCH_VALUES_DEC))
 
 
+def int2byte(i):
+    return bytes((i),)
+
```
Comment (wbarnha) [inline benchmarks/varint_speed.py:70]: > Forced pushed, let's see if I missed anything. I'm not 100% happy with the crimes I committed in `benchmarks/variant_speed.py`, but I also didn't want to completely rewrite it to remove its reliance on the Python 2 construct of int2byte.

It happens, it's a sacrifice that sometimes must be made to keep the ball rolling 😅

**Review by wbarnha — APPROVED** [review]
Tests pass, so I'm satisfied. LGTM!

**Pull request merged**